test/mpi/threads/perf/mt_rma_msgrate.c

```diff
@@ -34,10 +34,40 @@ enum RMA_OPs {
     OP_INVALID
 };
 
+enum SYNC_MODE {
+    USE_WIN_FENCE,
+    USE_WIN_LOCK_ALL,
+    SYNC_INVALID,
+};
+
+#define OPEN_WINDOW()                                   \
+    if (sync_mode == USE_WIN_FENCE) {                   \
+        MPI_Win_fence(0, my_win);                       \
+    } else if (sync_mode == USE_WIN_LOCK_ALL) {         \
+        MPI_Win_lock_all(MPI_MODE_NOCHECK, my_win);     \
```

Comment:
If I don't use `MPI_MODE_NOCHECK` here, the test will deadlock or hang in some instances. There could be a bug in our lockall/unlockall protocol. I'm going to make an issue to look into it further.